quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -181,12 +181,22 @@ impl S3CompatibleObjectStorage {
     pub async fn from_uri(
         s3_storage_config: &S3StorageConfig,
         uri: &Uri,
+    ) -> Result<Self, StorageResolverError> {
+        let s3_client = create_s3_client(s3_storage_config).await;
+        Self::from_uri_and_client(s3_storage_config, uri, s3_client).await
+    }
+
+    /// Creates an object storage given a region, an uri and an S3 client.
+    pub async fn from_uri_and_client(
```

Comment:
I don't think it is worth having 2 constructors here, `from_uri` is only used in a few tests

Reply:
i removed `new()` (it's not used outside of `from_uri_and_client()`). To remove `from_uri()` we'd need to make  `create_s3_client` pub (the tests are in tests/, not src/, so they can only use the public api). I don't think it's worth the hassle